### PR TITLE
patch the getKey() proc to support blocking mode

### PR DESCRIPTION
**Pull request by ghost**
hi @johnnovak , the blocking mode getKey() costs less cpu, maybe it's a better choice than the async way in some cases, so i patch it.

### CODE REVIEW

**Review by johnnovak — CHANGES_REQUESTED** [review]
Could be handy but I don't like the implementation. `getKey` should just use `getch` from the terminal module internally instead.

File: illwill.nim [hunk 1]
```diff
@@ -542,13 +562,17 @@ proc illwillDeinit*() =
   resetAttributes()
   showCursor()
 
-proc getKey*(): Key =
+proc getKey*(blocking: bool = false): Key =
   ## Reads the next keystroke in a non-blocking manner. If there are no
```
Comment (johnnovak) [inline illwill.nim:566]: Update method comment and the general documentation everywhere.

**Review by johnnovak — CHANGES_REQUESTED** [review]

File: illwill.nim (hunk 1 above)
Comment (ghost) [reply illwill.nim:566]: @johnnovak ok, i will update the relate comments and the code when i am free :)

Comment (johnnovak) [reply illwill.nim:566]: Ok, saw my other comment too?

> Could be handy but I don't like the implementation. getKey should just use getch from the terminal module internally instead.

Comment (ghost) [reply illwill.nim:566]: @johnnovak  yeah, i got it

**Comment by enthus1ast** [comment]
too bad that this is not implemented yet :|

**Comment by ghost** [comment]
@johnnovak @enthus1ast i am stupid, i read the source code of `getKey`, i find you use select to get keyboard's input, it's fine, so this pull request is needless.

**Pull request closed without merging**